### PR TITLE
Add `service_name` to `OutputCoverallsJsonCoverage`, check if file exists in `OutputJsonCoverage`

### PR DESCRIPTION
**Pull request by markuspf**

### CODE REVIEW

**Comment by markuspf** [comment]
Note that this breaks the current coverage scripts in GAP

File: gap/profiling.gi [hunk 1]
```diff
@@ -787,7 +787,7 @@ function(data, outfile, jobid, pathtoremove)
 
     IO_Write(outstream, "{\n");
     IO_Write(outstream, Concatenation("\"service_job_id\": \"", jobid, "\",\n"));
-    IO_Write(outstream, "\"service_name\": \"travis-ci\",\n");
+    IO_Write(outstream, Concatenation("\"service_name\": \"", service_name, "\",\n");
```
Comment (fingolfin) [inline gap/profiling.gi]: There are additional key/value pairs we may need to set. So I suggest that if we break things anyway, we rather change this API as follows: drop the `jobid` and `service_name` arguments to this function, instead pass in an `opt` record. Then write each key/value pair of that opt record into the file. To keep things simple, let's restrict the "values" to strings (and perhaps ints).

Then we can also set the `service_pull_request` and `service_branch` keys and more, as "documented" (haha) here: https://github.com/lemurheavy/coveralls-ruby/blob/master/lib/coveralls/configuration.rb

Comment (fingolfin) [reply gap/profiling.gi]: I guess we could also allow `opt` to be optional, and in that case, try to "guess" the CI data from the environment, as done in the file I just linked to.

File: gap/profiling.gd [hunk 2]
```diff
@@ -89,17 +89,21 @@ DeclareGlobalFunction("OutputAnnotatedCodeCoverageFiles");
...
 #!   <A>cover</A> should either be a profile previously read by
 #!   <F>ReadLineByLineProfile</F>, or the filename of a profile which will first
 #!   be read with <F>ReadLineByLineProfile</F>.
-#!   <A>jobid</A> A string containing the travis jobid. We only support travis-ci at the moment.
-#!   <A>pathtoremove</A> the path to the tested repository;
-#!   this path will be removed from all filenames.
+#!   <A>pathtoremove</A> is the path to the tested repository;
+#!   this path prefix will be removed from all filenames in <A>cover</A>.
+#!   Finally, <A>opt</A> is a record. Its key/value pairs are directly inserted into
+#!   the produce JSON, in the form of a JSON dictionary. This can be used to
```
Comment (markuspf) [inline gap/profiling.gd]: nitpick: "produced"?

**Comment by markuspf** [comment]
The additional changes look good to me.

**Comment by fingolfin** [comment]
OK, then I hope @ChrisJefferson is willing to merge this soon

**Pull request merged**